src/absil/ilwrite.fs

```diff
@@ -426,28 +426,49 @@ type MethodDefKey(tidx: int, garity: int, nm: string, rty: ILType, argtys: ILTyp
     // argument types (only argument type count). This is very important, since
     // hashing these is way too expensive
     let hashCode = 
-       hash tidx 
-       |> combineHash (hash garity) 
-       |> combineHash (hash nm) 
-       |> combineHash (hash argtys.Length)
-       |> combineHash (hash isStatic)
+        hash tidx 
+        |> combineHash (hash garity) 
+        |> combineHash (hash nm) 
+        |> combineHash (hash argtys.Length)
+        |> combineHash (hash isStatic)
     member key.TypeIdx = tidx
     member key.GenericArity = garity
     member key.Name = nm
     member key.ReturnType = rty
     member key.ArgTypes = argtys
     member key.IsStatic = isStatic
     override x.GetHashCode() = hashCode
-    override x.Equals(obj: obj) = 
+    override x.Equals(obj: obj) =
         match obj with 
-        | :? MethodDefKey as y -> 
-            tidx = y.TypeIdx && 
-            garity = y.GenericArity && 
-            nm = y.Name && 
-            // note: these next two use structural equality on AbstractIL ILType values
-            rty = y.ReturnType && 
-            List.lengthsEqAndForall2 (fun a b -> a = b) argtys y.ArgTypes &&
-            isStatic = y.IsStatic
+        | :? MethodDefKey as y ->
+            // Compare method def keys rearranged slightly to improve debugging experience.
+            let compareILTypes o1 o2 =
+                //
+                let getScopeAssemRef (o:ILType) =
+                    match o with
+                    | ILType.Value v ->
```

Comment:
Note:  What to do when both are primary assembly and they have different arefs?. 

I think that's okay, thaey should probably be considered a match .... but I'm not sure ... I'm also not sure how it would arise.